File: javascript/04_AboutLoops/digitSum_02.js
// ฟังก์ชันวิเคราะห์ตัวเลข
function digitSum(number) {
    // แปลงตัวเลขเป็นค่าบวกและเป็นสตริง
    let numStr = Math.abs(number).toString();
    let sum = 0;
    let products = 1;
    let digitArray = [];
    
    // วนลูปผ่านแต่ละตัวอักษรในสตริงที่แปลงจากตัวเลข
    for (let i = 0; i < numStr.length; i++) {
        // แปลงตัวอักษรเป็นตัวเลข
        const digit = parseInt(numStr[i]);
        // บวกตัวเลขเข้ากับผลรวม
        sum += digit;
        // คูณตัวเลขเข้ากับผลคูณ
        products *= digit;
        // เพิ่มตัวเลขลงในอาร์เรย์ของตัวเลข
        digitArray.push(digit);
    }
    
    // คืนค่าเป็นออบเจ็กต์ที่มีการวิเคราะห์ตัวเลข
    return {
        originalNumber: number, // ตัวเลขต้นฉบับ
        digits: digitArray, // อาร์เรย์ของตัวเลข
        sum: sum, // ผลรวมของตัวเลข
        product: products, // ผลคูณของตัวเลข
        numberOfDigits: numStr.length // จำนวนหลักของตัวเลข
    };
}

// ตัวอย่างการใช้งาน: วิเคราะห์ตัวเลข 12345 และพิมพ์ผลลัพธ์
console.log(digitSum(12345));

File: java/04_AboutLoops/digitSum_02.java
import java.util.Arrays;

public class digitSum_02 {
    public static class DigitAnalysis {
        public int originalNumber;
        public int numberOfDigits;
        public int[] digits;
        public int sum;
        public int product;

        @Override
        public String toString() {
            return "DigitAnalysis{" +
                   "originalNumber=" + originalNumber +
                   ", numberOfDigits=" + numberOfDigits +
                   ", digits=" + Arrays.toString(digits) +
                   ", sum=" + sum +
                   ", product=" + product +
                   '}';
        }
    }

    // 4. ฟังก์ชันวิเคราะห์ตัวเลข
    public static DigitAnalysis digitSum(int number) {
        DigitAnalysis analysis = new DigitAnalysis();
        analysis.originalNumber = number;
        
        // แปลงเป็นค่าบวกและเป็น String
        String numStr = String.valueOf(Math.abs(number));
        analysis.numberOfDigits = numStr.length();
        analysis.digits = new int[analysis.numberOfDigits];
        analysis.sum = 0;
        analysis.product = 1;

        for (int i = 0; i < numStr.length(); i++) {
            int digit = Character.getNumericValue(numStr.charAt(i));
            analysis.digits[i] = digit;
            analysis.sum += digit;
            analysis.product *= digit;
        }

        return analysis;
    }

    // ตัวอย่างการใช้งาน
    public static void main(String[] args) {
        // ทดสอบการวิเคราะห์ตัวเลข
        System.out.println(digitSum(12345));
    }
}